### PR TITLE
String as number

### PR DESCRIPTION
**Pull request by g4s8**
`new StringAsNumber("42")` instead of `Integer.parseInt("42")`

### CODE REVIEW

**Comment by 0crat** [comment]
@g4s8 I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/test/java/org/cactoos/text/StringAsNumberTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id$
+ * @since 0.2
+ */
+public final class StringAsNumberTest {
+
+    /**
+     * Test {@link Integer}.
+     */
+    @Test
```
Comment (yegor256) [inline src/test/java/org/cactoos/text/StringAsNumberTest.java]: @g4s8 see how I've done other tests: https://github.com/yegor256/cactoos/blob/master/src/test/java/org/cactoos/io/InputAsBytesTest.java I removed JavaDoc and added "reasons" to `assertThat`. I believe this is a better design. Also, see: https://github.com/teamed/qulice/issues/848

File: src/main/java/org/cactoos/text/StringAsNumber.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+package org.cactoos.text;
+
+/**
+ * String as {@link Number}.
+ *
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id$
+ * @since 0.2
+ */
+public final class StringAsNumber extends Number {
```
Comment (yegor256) [inline src/main/java/org/cactoos/text/StringAsNumber.java]: @g4s8 maybe `TextAsInteger` would be a more logical design in our case?

```java
int x = new TextAsInteger(new StringAsText("-158")).asValue()
```

**Comment by yegor256** [comment]
@g4s8 see some comments above

File: src/main/java/org/cactoos/text/StringAsNumber.java (hunk 2 above)
Comment (yegor256) [inline src/main/java/org/cactoos/text/StringAsNumber.java]: @g4s8 Also, we'll need `TextAsDouble`, `TextAsBoolean`, `TextAsLong`. Or maybe let's rename this one to `TextAsLong` and let users cast from `long` to `int` after parsing?

Comment (g4s8) [inline src/main/java/org/cactoos/text/StringAsNumber.java]: @yegor256 I think that casting is not elegant solution, so I've added `TextAsLong` and `TextAsInt` as different classes. Furthermore `Long` and `Integer` have separated reference cache for small values (`Long.LongCache.cache` and `Integer.IntegerCache.cache`)

**Comment by g4s8** [comment]
@yegor256 added `Scalar<>` implementations for `Long`, `Integer`, `Double`, `Float` and `Boolean`, fixed tests

**Comment by yegor256** [comment]
@rultor try to merge

**Pull request merged**